src/backend/logger/uploader.ts

```diff
@@ -0,0 +1,135 @@
+import { app } from 'electron'
+import { readFile } from 'fs/promises'
+import path from 'path'
+import { z } from 'zod'
+
+import { TypeCheckedStoreBackend } from '../electron_store'
+import { getLogFile } from './logfile'
+import { logError, logInfo, LogPrefix } from './logger'
+import { sendFrontendMessage } from '../main_window'
+
+import type { UploadedLogData } from 'common/types'
+
+const uploadedLogFileStore = new TypeCheckedStoreBackend('uploadedLogs', {
+  cwd: 'store',
+  name: 'uploadedLogs',
+  accessPropertiesByDotNotation: false
+})
+
+async function sendRequestToApi(formData: FormData, url = 'https://0x0.st') {
+  return fetch(url, {
+    body: formData,
+    method: 'post',
+    headers: {
+      'User-Agent': `HeroicGamesLauncher/${app.getVersion()}`
+    }
+  }).catch((err) => {
+    logError([`Failed to send data to 0x0.st:`, err], LogPrefix.LogUploader)
+    return null
+  })
+}
+
+/**
+ * Uploads the log file of a game / runner / Heroic to https://0x0.st
+ * @param name See {@link UploadedLogData.name}
+ * @param appNameOrRunner Used to get the log file path. See {@link getLogFile}
+ * @returns `false` if an error occurred, otherwise the URL to the uploaded file and {@link UploadedLogData}
+ */
+async function uploadLogFile(
+  name: string,
+  appNameOrRunner: string
+): Promise<false | [string, UploadedLogData]> {
+  const fileLocation = getLogFile(appNameOrRunner)
+  const filename = path.basename(fileLocation)
+
+  const fileContents = await readFile(fileLocation)
```

Comment:
we should probably add some check for file size, some games print too much output and we don't want to read a 1GB file in memory and also try to upload it

the site seems to have a limit of `Maximum file size: 512.0 MiB` but I'd pick a way smaller number (if they share a 10mb file it's probably full of useless information)

Reply:
I've now limited the file size to 10MiB